### DISCUSSION TITLE
imageWithLogo error

### DISCUSSION THREAD

**Discussion by andrtechno** [chillerlan/php-qrcode#78] [Q&A]
v4.3.1
imageWithLogo

Fatal error: Uncaught Error: Class 'chillerlan\QRCodeExamples\QRImageWithLogo' not found

php 7.4

**Comment by codemasher** [comment]
Have you run `composer install`? Any other details? See also: #74 and #75

**Reply by andrtechno** [reply]
1. composer require chillerlan/php-qrcode 4.3.1
2. edit a file INDEX.php copied the contents of the example imageWithLogo.php into it changed the paths and other data in it and opened it in the browser

also tried creating my class I also tried to create my own MyQRImageWithLogo.php class and copied what you have in this file into it, and changed the name spice to your own. and me in the INDEX.php namespace
`Fatal error: Uncaught Error: Class 'app\components\MyQRImageWithLogo' not found
`
the project on which Ii installed yii2 framework

**Comment by codemasher** [comment]
The examples are only available in the dev namespace. If you installed the package via `composer require` this is not available, you will need to install it with `composer require-dev`. https://getcomposer.org/doc/03-cli.md#command-line-interface-commands and https://getcomposer.org/doc/03-cli.md#require

**Comment by andrtechno** [comment]
1. remove chillerlan/php-qrcode
2. composer require --dev chillerlan/php-qrcode 4.3.1

main composer.json
```
    "require-dev": {
...
        "chillerlan/php-qrcode": "4.3.1"
    },
```

problem still exists

**Comment by codemasher** [comment]
Yes, this is how composer works. The examples are only available as dev requirements because they should never be available in production. If you want to use them, copy them over to your environment and change the namespace.

**Comment by codemasher** [comment]
For further inquiries I'd recommend [StackOverflow](https://stackoverflow.com/questions/tagged/php) or the [PHPhelp](https://www.reddit.com/r/PHPhelp/) subreddit because none of this is related to this library but modular PHP fundamentals (composer, autoloading).

**Comment by andrtechno** [comment]
QRImageWithLogoTest.php
```
<?php

namespace app\components\myexamples;

use chillerlan\QRCode\Output\{QRCodeOutputException, QRImage};

use function imagecopyresampled, imagecreatefrompng, imagesx, imagesy, is_file, is_readable;

class QRImageWithLogoTest extends QRImage{

... content copied from original

}

```

index.php
```
<?php
require_once __DIR__.'/vendor/autoload.php';

use chillerlan\QRCode\{QRCode, QROptions};

$data = 'https://www.youtube.com/watch?v=DLzxrzFCyOs&t=43s';

class LogoOptions extends QROptions{
    // size in QR modules, multiply with QROptions::$scale for pixel size
    protected int $logoSpaceWidth;
    protected int $logoSpaceHeight;
}

$options = new LogoOptions;

$options->version          = 7;
$options->eccLevel         = QRCode::ECC_H;
$options->imageBase64      = false;
$options->logoSpaceWidth   = 13;
$options->logoSpaceHeight  = 13;
$options->scale            = 5;
$options->imageTransparent = false;



$qrOutputInterface = new \app\components\myexamples\QRImageWithLogoTest($options, (new QRCode($options))->getMatrix($data));

// dump the output, with an additional logo
echo $qrOutputInterface->dump(null, __DIR__.'/../logo.png');
```

and it makes no difference where to be
in require or require-dev

**Comment by codemasher** [comment]
Look, i'm not here to solve your autoloading issues, nor to educate you about composer works - many other people have done this before and it works perfectly fine.

**Comment by andrtechno** [comment]
Okey
